File: CS/HowToImplementDetailViewItem.Module.Win/ClassIconDetailItem.cs
using System;
using System.Drawing;
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Editors;
using DevExpress.ExpressApp.Utils;
using DevExpress.ExpressApp.Win.Editors;

namespace HowToImplementDetailViewItem.Module.Win {

	[DetailViewItemName("ClassIcon")]
	public class ClassIconDetailItem : DetailViewItem {
		public ClassIconDetailItem(Type classType, DictionaryNode info)
			: base(info, classType) {
		}
		protected override object CreateControlCore() {
			ResizablePictureBox imageControl = new ResizablePictureBox();
			string imageName = Info.Parent.Parent.GetAttributeValue("ImageName");
			ImageInfo imageInfo = ImageLoader.Instance.GetLargeImageInfo(imageName);
			if(imageInfo.IsEmpty) {
				imageControl.Visible = false;
			}
			else {
				Image image = imageInfo.Image;
				imageControl.Image = image;
				imageControl.Width = image.Width;
				imageControl.Height = image.Height;
			}
			return imageControl;
		}
	}
}

File: VB/HowToImplementDetailViewItem.Module.Win/ClassIconDetailItem.vb
Imports System
Imports System.Drawing
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Editors
Imports DevExpress.ExpressApp.Utils
Imports DevExpress.ExpressApp.Win.Editors
<DetailViewItemName("ClassIcon")> _
Public Class ClassIconDetailItem
   Inherits DetailViewItem
   Public Sub New(ByVal classType As Type, ByVal info As DictionaryNode)
      MyBase.New(classType, info)
   End Sub
   Protected Overrides Function CreateControlCore() As Object
      Dim imageControl As ResizablePictureBox = New ResizablePictureBox()
      Dim imageName As String = Info.Parent.Parent.GetAttributeValue("ImageName")
      Dim imageInfo As ImageInfo = ImageLoader.Instance.GetLargeImageInfo(imageName)
      If imageInfo.IsEmpty Then
         imageControl.Visible = False
      Else
         Dim image As Image = imageInfo.Image
         imageControl.Image = image
         imageControl.Width = image.Width
         imageControl.Height = image.Height
      End If
      Return imageControl
   End Function
End Class




